treq/testing.py

```diff
@@ -205,3 +207,218 @@ def __init__(self, resource):
                 function = _reject_files(function)
 
             setattr(self, function_name, function)
+
+
+class StringStubbingResource(Resource):
+    """
+    A resource that takes a callable with 5 parameters
+    ``(method, url, params, headers, data) -> (code, headers, body)``
```

Comment:
Minor nitpick: this says "takes 5 parameters" but then uses this ad-hoc `->` notation to say that it returns some stuff too.  I would recommend just "```a callable that takes ``(method, url, params, headers, data)`` and returns ``(code, headers, body)``.```